client/web/compose/src/components/Public/Record/BulkEdit/index.vue

```diff
@@ -0,0 +1,207 @@
+              </b-button>
+            </b-input-group-append>
+          </b-input-group>
+        </b-form-group>
+
+        <div slot="footer">
+          <b-button
+            variant="outline-dark"
+            class="float-left"
+            @click="onReset"
+          >
+            {{ $t('recordList.bulkRecord.reset') }}
+          </b-button>
+
+          <b-button
+            variant="dark"
+            class="float-right"
+            :disabled="!fields.length"
+            @click="onSave()"
+          >
+            {{ $t('general.label.save') }}
+          </b-button>
+        </div>
+      </b-card>
+    </b-modal>
+  </div>
+</template>
+
+<script>
+import FieldEditor from 'corteza-webapp-compose/src/components/ModuleFields/Editor'
+import { VueSelect } from 'vue-select'
+import { validator, compose } from '@cortezaproject/corteza-js'
+import calculatePosition from 'corteza-webapp-compose/src/mixins/vue-select-position'
+
+export default {
+  i18nOptions: {
+    namespaces: 'block',
+  },
+
+  name: 'BulkEdit',
+
+  components: {
+    VueSelect,
+    FieldEditor,
+  },
+
+  mixins: [
+    calculatePosition,
+  ],
+
+  props: {
+    namespace: {
+      type: compose.Namespace,
+      required: true,
+    },
+    module: {
+      type: compose.Module,
+      required: true,
+    },
+  },
+
+  data () {
+    return {
+      fieldName: null,
+      currentField: null,
+      showModal: false,
+      selectedFields: [],
+      fields: [],
+      mock: {
+        show: true,
+        namespace: undefined,
+        module: undefined,
+        field: undefined,
+        record: undefined,
+        errors: new validator.Validated(),
+      },
+      render: true,
+    }
+  },
+
+  computed: {
+    fieldOptions () {
+      return this.module.fields.map(({ name, label }) => ({ name, label: label || name }))
+    },
+  },
+
+  created () {
+    const module = JSON.parse(JSON.stringify(this.module || {}))
+
+    this.mock.namespace = this.namespace
+    this.mock.module = module
+    this.mock.record = new compose.Record(this.mock.module, {})
+  },
+
+  methods: {
+    onModalHide () {
+      this.showModal = false
+    },
+
+    onClose () {
+      this.showModal = false
+    },
+
+    getField (name = '') {
+      const field = name ? this.mock.module.fields.find(f => f.name === name) : undefined
+
+      return field ? { ...field } : undefined
+    },
+
+    filterFieldOptions () {
+      const selectedFields = this.fields.map(({ name }) => name)
+      return this.module.fields.filter(({ name }) => !selectedFields.includes(name))
+    },
+
+    getOptionLabel ({ kind, label, name }) {
+      return label || name || kind
+    },
+
+    addField () {
+      this.fields.push(this.currentField)
+      this.currentField = null
+    },
+
+    onReset () {
+      this.mock.record = new compose.Record(this.module.clone(), {})
+      this.fields = []
+    },
+
+    onSave () {
+      this.$emit('record-update', this.mock.record)
```

Comment:
You should do the processing and api interaction here and after that emit an event that will refresh the record list